// ########################################################################
// Copyright 2017 loopchain authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// ########################################################################
//
// A proto file for gRPC interface.

syntax = "proto2";


// Peer Outer Service for Outer Client or other Peer
service PeerService {
    rpc GetStatus (StatusRequest) returns (StatusReply) {}
}

// Data for RadioStation
message StatusRequest {
    required string request = 1;
    optional string channel = 2;
}

message StatusReply {
    required string status = 1;
    required int32 block_height = 2;
    required int32 total_tx = 3;
    optional int32 unconfirmed_block_height = 4;
    optional int32 is_leader_complaining = 5;
    optional string peer_id = 6;
}
